packages/cli/src/nonInteractiveCli.test.ts

```diff
@@ -1008,4 +1008,265 @@ describe('runNonInteractive', () => {
     );
     expect(processStdoutSpy).toHaveBeenCalledWith('file.txt');
   });
+
+  describe('signal handling', () => {
+    it('should register SIGINT and SIGTERM handlers', async () => {
```

Comment:
this is worth an integration test as well. otherwise I would be impressed if it all just works seamlessly on windows.

Reply:
Will be addressed in a follow up PR